src/OnlineSales/Controllers/StorageController.cs

```diff
@@ -0,0 +1,104 @@
+﻿// <copyright file="StorageController.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using System.ComponentModel.DataAnnotations;
+using System.Security.Cryptography;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.EntityFrameworkCore;
+using OnlineSales.Data;
+using OnlineSales.DTOs;
+using OnlineSales.Entities;
+using OnlineSales.Helpers;
+
+namespace OnlineSales.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    public class StorageController : ControllerBase
+    {
+        private readonly PgDbContext pgDbContext;
+
+        public StorageController(PgDbContext pgDbContext)
+        {
+            this.pgDbContext = pgDbContext;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(typeof(MediaDetailsDto), StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status422UnprocessableEntity)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult> Post([FromForm] FileCreateDto fileCreateDto)
+        {
+            var provider = new FileExtensionContentTypeProvider();
+
+            fileCreateDto.ScopeUid = fileCreateDto.ScopeUid.TrimStart('/');
+
+            var incomingFileName = fileCreateDto.File!.FileName.ToTranslit().Slugify()!;
+            var incomingFileExtension = Path.GetExtension(fileCreateDto.File!.FileName);
+            var incomingFileSize = fileCreateDto.File!.Length; // bytes
+            var incomingFileMimeType = string.Empty;
+
+            if (!provider.TryGetContentType(incomingFileName, out incomingFileMimeType))
+            {
+                ModelState.AddModelError("FileName", "Unsupported MIME type");
+
+                throw new InvalidModelStateException(ModelState);
+            }
+
+            using var fileStream = fileCreateDto.File.OpenReadStream();
+            var fileInBytes = new byte[incomingFileSize];
+            fileStream.Read(fileInBytes, 0, (int)fileCreateDto.File.Length);
+
+            var scopeAndFileExists = await pgDbContext!.StoredFiles!.Where(e => e.ScopeUid == fileCreateDto.ScopeUid && e.Name == incomingFileName!).ToArrayAsync();
+            if (!scopeAndFileExists.Any())
```

Comment:
What happens if there is file with the same ScopeUid and Name in the database?

Reply:
Fixed by adding check for exist record and update it when entity exists